File: boj/brute-force/1107.java

import java.io.*;

public class Main {

    static int goal;
    static boolean[] disables;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        // input & init
        goal = Integer.parseInt(br.readLine());
        int count = Integer.parseInt(br.readLine());

        String[] input;
        disables = new boolean[10];

        if (count != 0) {
            input = br.readLine().split(" ");
            for (int i = 0; i < input.length; i++) {
                disables[Integer.parseInt(input[i])] = true;
            }
        }

        int result = search();
        bw.write(result + "\n");
        bw.flush();
    }

    static int search() {
        int MIN = Math.abs(goal - 100);

        for (int cur = 0; cur <= 1000000; cur++) {
            int count = countOfPressingButton(cur);

            if (count == -1) continue;
            int pressingNumber = Math.abs(goal - cur);
            MIN = Math.min(count + pressingNumber, MIN);
        }

        return MIN;
    }

    static int countOfPressingButton(int num) {
        if (num == 0) {
            if (disables[num]) return -1;
            return 1;
        }

        int count = 0;
        while (num > 0) {
            if (disables[num % 10]) return -1;
            num /= 10;
            count++;
        }

        return count;
    }

}

File: boj/brute-force/1107.swift
import Foundation

let initialChannel = 100

var channel = [Bool](repeating: true, count: 10)
var targetChannel = Int()
var unavailableCnt = Int()
var unavailableChannel = [Int]()

func pressingButton( _ input: Int) -> Int {
    var number = input
    
    if number == 0 {
        if channel[number] {
            return 1
        }
        else { return -1 }
    }
    
    var count = 0
    while number > 0 {
        if !channel[number % 10] { return -1 }
        number /= 10
        count += 1
    }
    
    return count
}

func solution() {
    targetChannel = Int(readLine()!)!
    unavailableCnt = Int(readLine()!)!
    if unavailableCnt != 0 {
        unavailableChannel = readLine()!.split(separator: " ").map { Int(String($0))! }
    
        for ele in unavailableChannel {
            channel[ele] = false
        }
    }
    
    var result = abs(targetChannel - 100)
    for i in 0...1000000 {
        
        let count = pressingButton(i)
        
        if count == -1 { continue }
        result = min(result, count + abs(targetChannel - i))
    }
    
    print(result)
}

solution()
